### ISSUE TITLE
DomConvert#viewToDom() conflicts with #mapViewToDom()

### ISSUE THREAD

**Issue by Reinmar** [ckeditor/ckeditor5-engine#993]
As far as I understand, `viewToDom()` creates DOM elements based on view elements and `mapViewToDom()` maps already created ones using mappings.

So, let's rename `viewToDom()` to `createDomFromView()` and `domToView()` to `createViewFromDom()`. Or something equally clear.

**Comment by pjasiun** [comment]
Yes, `viewToDom()` is a method to create elements, so `createDomFromView()` is a better name.

**Comment by szymonkups** [comment]
`viewToDom` not only creates nodes but returns mapped elements if mapping exists. But the main purpose is to create DOM element from View element (it can be new or mapped one), so 👍 for `createDomFromView`.

**Comment by Reinmar** [comment]
Hm... I've got doubts now. Should `createSth()` indeed use mappings? This will be against SRP and against its name.

I'd try to find a different name or make that method only create new elements. The former may be tricky, but is the latter reasonable at all?

**Comment by scofalik** [comment]
I think that `viewToDom` is fine, as far as the name goes. Just make it clear that this is the endpoint that should be used. Maybe other methods should be private if `viewToDom` is all that is needed?

**Comment by Reinmar** [comment]
That was my question at some point too. Why do we have two methods?

I think I know the answer, but let's see if it's still valid. @pjasiun?

**Comment by szymonkups** [comment]
Also, both methods: `domToView` and `viewToDom` return element from mapping if one exists. I found it would be useful if this method could create a new set of DOM/view elements without taking anything from mapping. When solving issue in typing I needed to create view elements from current DOM (which was desynchronized with view at the moment) and got mapped elements instead of new ones. I was forced to use new instance on `DomConverter`.

**Comment by pjasiun** [comment]
That's fine for me to make`createDomFromView` always create a new element, without taking it from mapping as long as the code which uses it will not be far more complicated because of it.

**Comment by Reinmar** [comment]
I fell for that again, so rising the priority a bit.
